### PR TITLE
Stepper (Desktop > Mobile > Finish)

### PR DESCRIPTION
**Pull request by myrosvas**

### CODE REVIEW

File: components/Stepper.vue [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+          <PresetList ref="presets" />
+        </div>
+
+        <v-btn color="green darken-3" @click="changeStep(2)">
+          Next
+        </v-btn>
+      </v-stepper-content>
+
+      <v-stepper-content step="2">
+        <div class="stepper-content__wrapper mobile">
```
Comment (andriytereschuk) [inline components/Stepper.vue]: here `<slot></slot>`

parent:
`<Stepper>
          <ComponentsMenu ref="componentsMenu" />
          <Workspace :current-step="currentStep" @add="add" />
          <PresetList ref="presets" />
</Stepper>`

File: api/projects.js [hunk 2]
```diff
@@ -13,31 +13,66 @@ export default (req, res) => {
...
       created: '2012-04-22T18:25:43.511Z',
-      modified: '2013-04-22T18:25:43.511Z'
+      modified: '2013-04-22T18:25:43.511Z',
+      desktop: {
+        rows: []
+      },
+      mobile: {
+        rows: []
+      },
+      currentStep: 'desktop'
```
Comment (andriytereschuk) [inline api/projects.js]: `currentStep` redundant prop; it is "stepper" state

Comment (andriytereschuk) [reply api/projects.js]: use inside app only

File: components/PresetList.vue [hunk 3]
```diff
@@ -28,20 +28,28 @@
 </template>
 
 <script>
-import Preset from '~/components/Preset'
+import { mapState } from 'vuex'
 import { presets } from '~/config/preset.config'
+import { mobilePresets } from '~/config/mobilePreset.config'
```
Comment (andriytereschuk) [inline components/PresetList.vue]: hold mobile and desktop presets in 1 file

{
mobile: [], desktop: []
}

File: components/PresetList.vue [hunk 4]
```diff
@@ -28,20 +28,28 @@
...
-      dialog: false
+      dialog: false,
+      stepsEnum
     }
   },
   computed: {
+    ...mapState('project', ['project']),
     presets() {
-      return presets
+      return this.project.currentStep === this.stepsEnum.desktop
```
Comment (andriytereschuk) [inline components/PresetList.vue]: use vuex getter to get `currentStep` or pass it down from parent

Comment (andriytereschuk) [reply components/PresetList.vue]: then 
`presets() { return presets[currentStep]}`

File: components/Workspace.vue [hunk 5]
```diff
@@ -17,22 +17,24 @@
...
-
 export default {
   components: {
     Section,
     AddSection,
     draggable
   },
+  data: () => {
+    return {
+      stepsEnum
```
Comment (andriytereschuk) [inline components/Workspace.vue]: useless

File: plugins/vue-json-viewer.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import Vue from 'vue'
+import JsonViewer from 'vue-json-viewer'
+import '~/assets/scss/json-viewer.scss'
```
Comment (andriytereschuk) [inline plugins/vue-json-viewer.js]: probably useless
since you added your own stylesheet to assets

File: store/project.js [hunk 7]
```diff
@@ -80,8 +88,13 @@ export const actions = {
...
       commit('add', {
-        id: project.id,
-        rows: []
+        desktop: {
+          rows: []
+        },
+        mobile: {
+          rows: []
+        },
+        currentStep: stepsEnum.desktop
```
Comment (andriytereschuk) [inline store/project.js]: put `currentStep` somewhere else

File: store/project.js [hunk 8]
```diff
@@ -102,11 +115,23 @@ export const mutations = {
...
+      rows: [...state.project.desktop.rows, row]
+    }
+    const mobile = {
+      ...state.project.mobile,
+      rows: [...state.project.mobile.rows, row]
+    }
+
+    state.project.currentStep === stepsEnum.desktop
+      ? (state.project = { ...state.project, desktop })
+      : (state.project = { ...state.project, mobile })
```
Comment (andriytereschuk) [inline store/project.js]: wow
take it easy

there is a  generic way to set such things
by passing `currentStep` as a key

plz, optimize lines 118 - 129

File: store/project.js [hunk 9]
```diff
@@ -102,11 +115,23 @@ export const mutations = {
...
   },
   attachComponent(state, payload) {
-    const rows = state.project.rows.map((section) => {
+    const { currentStep } = state.project
+    let screenType = state.project[currentStep]
```
Comment (andriytereschuk) [inline store/project.js]: well
let think about the next thing
what to do if we attach a component to desktop then shall we add it to mobile as well?

File: store/project.js [hunk 10]
```diff
@@ -115,14 +140,15 @@ export const mutations = {
         ...section
       }
     })
-
-    state.project = {
-      rows,
-      ...state.project
+    screenType = {
```
Comment (andriytereschuk) [inline store/project.js]: not obvious with reusing that variable

File: store/project.js [hunk 11]
```diff
@@ -132,18 +158,37 @@ export const mutations = {
...
+      }
+    })
+  },
+  setCurrentStep(state, newStep) {
+    state.project.currentStep = newStep
   }
 }
 
 export const getters = {
   sections(state) {
```
Comment (andriytereschuk) [inline store/project.js]: same
make it simpler
`return state.project[step]rows`

File: plugins/vue-json-viewer.js (hunk 6 above)
Comment (myrosvas) [reply plugins/vue-json-viewer.js]: Here it is needed, if I comment it out, styles are not applied

**Pull request merged**